driver-core/src/main/com/mongodb/internal/operation/ClientBulkWriteOperation.java

```diff
@@ -772,40 +772,24 @@ ClientBulkWriteResult build(@Nullable final MongoException topLevelError, final
-                            AbstractClientNamespacedWriteModel writeModel = getNamespacedModel(models, writeModelIndex);
-                            if (writeModel instanceof ConcreteClientNamespacedInsertOneModel) {
-                                insertResults.put(
-                                        writeModelIndex,
-                                        new ConcreteClientInsertOneResult(insertModelDocumentIds.get(individualOperationIndexInBatch)));
-                            } else if (writeModel instanceof ConcreteClientNamespacedUpdateOneModel
-                                    || writeModel instanceof ConcreteClientNamespacedUpdateManyModel
-                                    || writeModel instanceof ConcreteClientNamespacedReplaceOneModel) {
-                                BsonDocument upsertedIdDocument = individualOperationResponse.getDocument("upserted", null);
-                                updateResults.put(
-                                        writeModelIndex,
-                                        new ConcreteClientUpdateResult(
-                                                individualOperationResponse.getInt32("n").getValue(),
-                                                //TODO-JAVA-6005 Previously, we did not provide a default value of 0 because the
-                                                // server was suppose to return nModified as 0 when no documents were changed.
-                                                // Due to server bug SERVER-113026, we must provide a default of 0 until we stop supporting
-                                                // server versions with this bug. When that happens, remove the default value for nModified.
-                                                individualOperationResponse.getInt32("nModified", new BsonInt32(0)).getValue(),
-                                                upsertedIdDocument == null ? null : upsertedIdDocument.get("_id")));
-                            } else if (writeModel instanceof ConcreteClientNamespacedDeleteOneModel
-                                    || writeModel instanceof ConcreteClientNamespacedDeleteManyModel) {
-                                deleteResults.put(
-                                        writeModelIndex,
-                                        new ConcreteClientDeleteResult(individualOperationResponse.getInt32("n").getValue()));
-                            } else {
-                                fail(writeModel.getClass().toString());
-                            }
+                        if (okFieldValue == 1) {
+                            collectSuccessfulIndividualOperationResult(
+                                    individualOperationResponse,
+                                    writeModelIndex,
+                                    individualOperationIndexInBatch, insertResults,
```

Comment:
Each other argument is on its own line, which suggests that `individualOperationIndexInBatch` and `insertResults` should also be on different lines.